pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -603,7 +612,11 @@ func (r *reconciler) reconcileConfigMaps(ctx context.Context, data reconcileData
 func (r *reconciler) reconcileSecrets(ctx context.Context, data reconcileData) error {
 	creators := []reconciling.NamedSecretCreatorGetter{
 		openvpn.ClientCertificate(data.openVPNCACert),
-		cloudcontroller.CloudConfig(data.cloudConfig),
+		cloudcontroller.CloudConfig(data.cloudConfig, resources.CloudConfigSecretName),
+	}
+
+	if r.cloudProvider == mctypes.CloudProviderVsphere {
```

Comment:
Tiny nit: it might be better to check is `data.csiCloudConfig != nil`. In that case, it might be easier to extend it for other providers in the future (if needed).